src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleGroupSyncMonitor.scala

```diff
@@ -132,28 +138,45 @@ class GoogleGroupSyncMonitorActor(
 
   override def receive = {
     case StartMonitorPass =>
-      logger.debug(s"Pulling messages off: $pubSubSubscriptionName")
+      logger.info(s"Pulling messages off: $pubSubSubscriptionName")
       // start the process by pulling a message and sending it back to self
       pubSubDao.pullMessages(pubSubSubscriptionName, 1).map(_.headOption) pipeTo self
 
     case Some(message: PubSubMessage) =>
-      logger.debug(s"received sync message: $message")
-      import Tracing._
-      trace("GoogleGroupSyncMonitor-PubSubMessage") { span =>
+      logger.info(s"received sync message: $message")
+
+      traceIO("GoogleGroupSyncMonitor-PubSubMessage", SamRequestContext()) { samRequestContext =>
+        // every 15s reset the deadline to 60s in the future, canceled when IO is done
+        // 15s so we get a few tries at extending before deadline occurs
+        val deadlineExtender = system.scheduler.scheduleAtFixedRate(Duration.Zero, 15 seconds, self, ExtendDeadline(message.ackId, 60, samRequestContext))
+
         val groupId: WorkbenchGroupIdentity = parseMessage(message)
+        samRequestContext.parentSpan.foreach(
+          _.putAttributes(
+            Map("groupId" -> AttributeValue.stringAttributeValue(groupId.toString), "messageId" -> AttributeValue.stringAttributeValue(message.ackId)).asJava
+          )
+        )
         groupSynchronizer
           .synchronizeGroupMembers(
             groupId,
-            samRequestContext = SamRequestContext(Option(span))
-          ) // Since this is an internal pub/sub call, we have to start a new SamRequestContext.
-          .toTry
-          .map(sr => sr.fold(t => FailToSynchronize(t, message.ackId), x => ReportMessage(x, message.ackId))) pipeTo self
-      }
+            samRequestContext = samRequestContext
+          )
+          .redeem(t => FailToSynchronize(t, message.ackId), x => ReportMessage(x, message.ackId))
+          .guarantee(IO(deadlineExtender.cancel()))
+      }.unsafeToFuture() pipeTo self
+
+    case ExtendDeadline(ackId, deadlineSeconds, samRequestContext) =>
+      traceIOWithContext("extendDeadline", samRequestContext) { _ =>
+        IO.fromFuture(IO(pubSubDao.extendDeadlineById(pubSubSubscriptionName, Seq(ackId), deadlineSeconds))).timeout(10 seconds).handleError { t =>
```

Comment:
`IO.fromFuture, but being called on something wrapped in an `IO`? My brain hurts... Care to explain?

Reply:
https://typelevel.org/cats-effect/docs/2.x/datatypes/io#fromfuture

Reply:
Ah, `extendDeadlineById` returns a `Future`. Missed that. 

Side-grump: Its annoying that we mix `IO` and `Future` everywhere...

Reply:
I agree. The google part of workbench-libs we use is all `Future`y. The google2 part is more compatible with `IO` but then requires swallowing `fs2.Stream`s and porting the google directory doa (and maybe others)